### DISCUSSION TITLE
Add file to MediaField

### DISCUSSION THREAD

**Discussion by vitalybrandes** [OrchardCMS/OrchardCore#9691] [Q&A] [answered]
I am programmably update ContentItem with adding file to MediaField.

```
 var result = await _contentManager.GetAsync(ContentItemId);
 if (result != null)
{
     result.Alter<Printer>(nameof(Printer), part =>
     {
         part.BackupFile = new MediaField { Paths = new string[] { FilePath } };
     });
     await _contentManager.UpdateValidateAndCreateAsync(result, VersionOptions.Published);
}
```
 Actually, i found that that  this replace the existing data from part.BackupFile to new one, instead add.
the BackupFile  in Media Filed with Multiple files allowed.

What i am doing wrong?

**Comment by lampersky** [comment, marked as answer]
`part.BackupFile = new MediaField { Paths = new string[] { FilePath } };`
This line replaces current **Paths** value, with single item array.
Maybe try to read **Paths** first, append additional item and save it back.
